flax/linen/combinators.py

```diff
@@ -0,0 +1,44 @@
+"""Combinators of modules, such as a Sequential."""
+
+from typing import Callable, Sequence
+
+from flax.linen.module import Module
+
+class Sequential(Module):
+  """Applies a linear chain of Modules.
+
+  Meant to be used only for the simple case of fusing together callables where
+  the input of a particular module/op is the output of the previous one.
+
+  Modules will be applied in the order that they are passed in the constructor.
+
+  The apply() method of Sequential accepts any input and forwards it to the
+  first module it contains. It chains the output sequentially to the input of
+  the next module and returns the output of the final module.
+
+  Example usage:
+
+    class Foo(nn.Module):
+        feature_sizes: List[int]
```

Comment:
Nit: for annotating arguments abstract collection types are preferred (see [here](https://docs.python.org/3/library/typing.html#typing.List)]

```suggestion
        feature_sizes: Sequence[int]
```